### PR TITLE
Don't overwrite blank (but non-empty) dominant values during mergeXpp3Dom

### PR DESCRIPTION
**Pull request by kwin**
Compare with #212

### CODE REVIEW

File: src/main/java/org/codehaus/plexus/util/xml/Xpp3DomUtils.java [hunk 1]
```diff
@@ -282,6 +282,6 @@ public static boolean isNotEmpty( String str )
 
     public static boolean isEmpty( String str )
     {
-        return ( str == null || str.trim().length() == 0 );
+        return ( str == null || str.length() == 0 );
```
Comment (michael-o) [inline src/main/java/org/codehaus/plexus/util/xml/Xpp3DomUtils.java:285]: This is fatally UGLY. P-U has StringUltils#isEmtpy(). Why not remove and use that?

**Review by michael-o — CHANGES_REQUESTED** [review]

File: src/main/java/org/codehaus/plexus/util/xml/Xpp3DomUtils.java (hunk 1 above)
Comment (kwin) [reply src/main/java/org/codehaus/plexus/util/xml/Xpp3DomUtils.java:285]: It is easier to spot the difference when keeping the other conditions as is. I would rather cleanup in a subsequent commit. WDYT?

**Comment by kwin** [comment]
> I wonder what will fail with current downstream consumer tests

Any particular tests you have in mind? I will try to run Maven ITs against this change.

File: src/main/java/org/codehaus/plexus/util/xml/Xpp3DomUtils.java (hunk 1 above)
Comment (michael-o) [reply src/main/java/org/codehaus/plexus/util/xml/Xpp3DomUtils.java:285]: Agreed. Leave a comment about removal.

**Comment by michael-o** [comment]
> > I wonder what will fail with current downstream consumer tests
> 
> Any particular tests you have in mind? I will try to run Maven ITs against this change.

Run this patched version against Core plugins ITs.

**Comment by michael-o** [comment]
I'd like to run tests on plugins. Does it suffice to update Maven Core with this patch?

**Comment by kwin** [comment]
I modified Maven master to define
`<plexusUtilsVersion>3.4.3-SNAPSHOT</plexusUtilsVersion>` in the reactor pom.xml, build it and ran the integration test successfully against that distro.

**Comment by michael-o** [comment]
> I modified Maven master to define `<plexusUtilsVersion>3.4.3-SNAPSHOT</plexusUtilsVersion>` in the reactor pom.xml, build it and ran the Maven Core integration tests successfully against that distro.

Good, I will repeat with several core plugin ITs with 3.9.0-SNAPSHOT tomorrow.

Will this PR completely cover the downstream Maven issue?

**Comment by kwin** [comment]
> Will this PR completely cover the downstream Maven issue?

No, it just allows to apply a workaround by giving a blank value in the plugin configuration to disable the default handling (FTR: the downstream issue is https://issues.apache.org/jira/browse/MNG-6434).

File: src/main/java/org/codehaus/plexus/util/xml/Xpp3DomUtils.java (hunk 1 above)
Comment (kwin) [reply src/main/java/org/codehaus/plexus/util/xml/Xpp3DomUtils.java:285]: This is a public method, so probably this needs deprecation before removing it finally.

Comment (michael-o) [reply src/main/java/org/codehaus/plexus/util/xml/Xpp3DomUtils.java:285]: Makes sense. The impl can be exchanged for StringUtils for consistency reasons.

Comment (kwin) [reply src/main/java/org/codehaus/plexus/util/xml/Xpp3DomUtils.java:285]: I added https://github.com/codehaus-plexus/plexus-utils/issues/215 to track that.

**Pull request merged**